src/coreclr/gc/gc.cpp

```diff
@@ -39520,6 +40354,32 @@ void gc_heap::verify_regions (int gen_number)
 
     while (seg_in_gen)
     {
+        if (can_verify_gen_num)
+        {
+            if (heap_segment_gen_num (seg_in_gen) != min (gen_number, max_generation))
+            {
+                dprintf (REGIONS_LOG, ("h%d gen%d region %Ix(%Ix) gen is %d!", 
```

Comment:
should the log also mention this as a FATAL condition?

Reply:
we don't mention it's fatal in the messages for FATAL_GC_ERROR()s.... it should be pretty obvious when this is the last msg I think...